tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/planner_utils.h

```diff
@@ -101,6 +104,64 @@ bool isValidState(const tesseract_kinematics::ForwardKinematics::ConstPtr& robot
   return !(robot_config != RobotConfig::FUT && robot_config != RobotConfig::NUT);
 }
 
+/**
+ * @brief Get the profile string taking into account defaults and profile remapping
+ * @param profile The requested profile name in the instructions
+ * @param planner_name Used to look up if there are remappings available
+ * @param profile_remapping Remapping used to remap a profile name based on the planner name
+ * @param default_profile Default = DEFAULT. This is set if profile.empty()
+ * @return The profile string taking into account defaults and profile remapping
+ */
+inline std::string getProfileString(const std::string& profile,
+                                    const std::string& planner_name,
+                                    const PlannerProfileRemapping& profile_remapping,
+                                    std::string default_profile = "DEFAULT")
+{
+  std::string results = profile;
+  if (profile.empty())
+    results = std::move(default_profile);
+
+  // Check for remapping of profile
+  auto remap = profile_remapping.find(planner_name);
+  if (remap != profile_remapping.end())
+  {
+    auto p = remap->second.find(profile);
+    if (p != remap->second.end())
+      results = p->second;
+  }
+  return results;
+}
+
+template <typename ProfileType>
+std::shared_ptr<ProfileType>
+getProfile(const std::string& profile, const std::unordered_map<std::string, std::shared_ptr<ProfileType>>& profile_map, std::string default_profile = "DEFAULT")
```

Comment:
Maybe switch this to where the user passes in the actual default profile instead of the string?

Reply:
Ok. I made this change

Reply:
I like the change, but maybe we should put the "DEFAULT" in a static const variable that gets used everywhere?

Reply:
You mean for each planner store the default profile in a separate const variable? We would have to pass them into the problem generators for ompl, trajopt, and descartes. Ideally, the default in getProfile would never be used and it would just use whatever has been set as DEFAULT. They are only used in the the raster managers because they currently replace the profile_map of the planners rather than inserting the new entries.

Reply:
Sorry, I meant add the line below some where in a base class that all use then use this variable instead of `"DEFAULT"`.

```
static const std::string DEFAULT_PROFILE_KEY = "DEFAULT";
```

Reply:
I added it to MotionPlanner.

Reply:
Maybe we should move somewhere into the command lanaguage which is the lowest level and then it could be used a few places in the command language along with in motion planning?

Reply:
Where would you want it? The Instruction?

Reply:
Maybe just create a types.h or constants.h in the same directory as composite_instruction.h?